pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -527,6 +527,8 @@ func doNetworkInstanceCreate(ctx *zedrouterContext,
 			status.BridgeIPAddr, &status.NetworkInstanceConfig,
 			hostsDirpath, status.BridgeIPSets, status.Ipv4Eid)
 		startDnsmasq(bridgeName)
+
+		go DNSMonitor(bridgeName, bridgeNum)
```

Comment:
We will need a handle (like a channel) to send a signal to go -routing when network instance gets deleted. When network instance is deleted, we should make sure that the corresponding go-routine also exits.

Reply:
Ok. I get it now. You have a global map that had this channel per network instance.

Reply:
right.